src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -248,6 +248,19 @@ private void SwitchToSpeedTestPhase(object _)
             Interlocked.Increment(ref state.SpeedTestMode);
         }
 
+        internal void DisableFastestNodeReadBalance()
+        {
+            if (_updateFastestNodeTimer != null)
```

Comment:
shouldn't the code in this method be similar to the ScheduleSpeedTest one?

```csharp
        internal void DisableFastestNodeReadBalance()
        {
            if (_updateFastestNodeTimer == null)
                return;

            lock (_timerCreationLocker)
            {
                if (_updateFastestNodeTimer == null)
                    return;

                _updateFastestNodeTimer.Dispose();
                _updateFastestNodeTimer = null;

                var state = _state;
                Interlocked.Exchange(ref state.SpeedTestMode, 0);
                Array.Clear(state.FastestRecords, index: 0, state.FastestRecords.Length);
            }
        }
```

Reply:
Done.